Life-Expectancy-and-GDP-Starter/.ipynb_checkpoints/life_expectancy_gdp-checkpoint.ipynb

```diff
@@ -4,25 +4,1328 @@
+      "text/plain": [
+       "<Figure size 640x480 with 1 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "# Create a Figure\n",
+    "fig = plt.figure(figsize = (10, 10))\n",
+    "plt.subplots_adjust(hspace = .5)\n",
+    "\n",
+    "# Create the First Subplot, Chile\n",
+    "ax1 = plt.subplot(3, 2, 1) # 3 Rows, 2 Columns, 1st Subplot\n",
+    "plt.hist(chile[\"Life\"])\n",
+    "plt.title(\"Chile's Life Expectancy\")\n",
+    "plt.xlabel(\"Years\")\n",
+    "plt.ylabel(\"Frequency\")\n",
+    "\n",
+    "# Create the Second Subplot, China\n",
+    "ax2 = plt.subplot(3, 2, 2)\n",
+    "plt.hist(china[\"Life\"])\n",
+    "plt.title(\"China's Life Expectancy\")\n",
+    "plt.xlabel(\"Years\")\n",
+    "plt.ylabel(\"Frequency\")\n",
+    "\n",
+    "# Create the Third Subplot, Germany\n",
+    "ax3 = plt.subplot(3, 2, 3)\n",
+    "plt.hist(germany[\"Life\"])\n",
+    "plt.title(\"Germany's Life Expectancy\")\n",
+    "plt.xlabel(\"Years\")\n",
+    "plt.ylabel(\"Frequency\")\n",
+    "\n",
+    "# Create the Fourth Subplot, Mexico\n",
+    "ax4 = plt.subplot(3, 2, 4)\n",
+    "plt.hist(mexico[\"Life\"])\n",
+    "plt.title(\"Mexico's Life Expectancy\")\n",
+    "plt.xlabel(\"Years\")\n",
+    "plt.ylabel(\"Frequency\")\n",
+    "\n",
+    "# Create the Fifth Subplot, USA\n",
+    "ax5 = plt.subplot(3, 2, 5)\n",
+    "plt.hist(usa[\"Life\"])\n",
+    "plt.title(\"USA's Life Expectancy\")\n",
+    "plt.xlabel(\"Years\")\n",
+    "plt.ylabel(\"Frequency\")\n",
+    "\n",
+    "# Create the Sixth Subplot, Zimbabwe\n",
+    "ax6 = plt.subplot(3, 2, 6)\n",
+    "plt.hist(zimbabwe[\"Life\"])\n",
+    "plt.title(\"Zimbabwe's Life Expectancy\")\n",
+    "plt.xlabel(\"Years\")\n",
+    "plt.ylabel(\"Frequency\")\n",
+    "\n",
+    "# Show and Close the Plot\n",
+    "plt.show()\n",
+    "plt.close()\n",
+    "\n",
+    "##########\n",
+    "# Create a histogram of All Countries' Life Expectancy\n",
+    "plt.hist(gdp_data[\"Life\"])\n",
+    "\n",
+    "# Create labels for the plot\n",
+    "plt.title(\"All Countries' Life Expectancy\")\n",
+    "plt.xlabel(\"Years\")\n",
+    "plt.ylabel(\"Frequency\")\n",
+    "\n",
+    "# Show and close the plot\n",
+    "plt.show()\n",
+    "plt.close()"
+   ]
```

Comment:
Suggestion for improvement:Instead of hardcoding each  subplot, a for loop can be created, see the example below:
```
#plot histograms of GPD for each country 
plt.figure(figsize=(10,10))
#Counter to change the subplot. 
counter = 1
for items in gdp_data.Country.unique():
    country_data=gdp_data[gdp_data.Country == items]
    plt.subplot(3,3,counter)
    sn.histplot(data=country_data, x='GDP', color='green')
    plt.title(f'GDP of {items}', fontsize=9)
    plt.xlabel('GPD in USD', fontsize=9)
    plt.ylabel('Life Expectancy', fontsize =9)
    plt.xticks(fontsize=9)
    plt.yticks(fontsize=9)
    counter = counter + 1
#Adjust spaces between plots. 
plt.subplots_adjust(wspace = 0.4, hspace= 0.4, bottom = 0.2)
plt.show()
plt.clf()
```

Reply:
Good idea. I'll look into making those changes to make efficient code.

Reply:
I updated the figure to use a for loop. Good suggestion